ci.md

```diff
@@ -0,0 +1,201 @@
+what checks are required for each library.
+
+## Interpreting results
+
+On the GitHub UI, checks can be:
+
+* ✅: Passed, all good!
+* 🟡: Pending, results haven't been received yet.
+* ❌: Failed, something is wrong.
+
+Depending on the library and on the build queue, checks can be in a pending
+🟡 state from a couple of minutes up to an entire day. If a build is in this
+state for too long, there may be some issue with infrastructure and the
+build didn't run or didn't report back.
+
+Failing builds ❌ usually require more investigation:
+
+* If the build is `Required`, something is definitely wrong and must be fixed
+  before merging the pull request.
+* If the build is not required, investigation is needed to find out if the
+  failure is pre-existing or if it is being introduced by the pull request.
+
+Clicking on the `Details` button of a failing build takes you to the build so
+that you can inspect it.
+
+Builds can fail for a variety of reasons, for example:
+
+* The code failed to compile. This should never be accepted, even in non-required
+  checks, and must be fixed.
+    * Jenkins: When there's a compilation failure, the build is marked red 🔴.
+               On the build's page, click on `Console Output` then scroll until
+               you find the compilation errors under the `compiling` section.
+    * Actions: When there's a compilation failure, the build is marked red ❌.
+               On the build logs, the compilation failure should be under the
+               `make` collapsible.
+* There are test failures. Test failures that aren't pre-existing must be fixed.
+    * Jenkins: When there are test failures, the build is marked yellow 🟡.
+               On the build's page, the failing tests are listed under
+               `Test Result`. You can also check the `Console Output` for the
+               full test logs.
+    * Actions: When there are test failures, the build is marked red ❌.
+               On the build logs, the test failures should be under the
+               `make test` collapsible.
+    * Check which tests are known failures by searching for the test name on
+      issues.
+* There are warnings. Warnings that aren't pre-existing must be fixed.
+    * Jenkins: When there are warnings, the build is marked yellow 🟡.
+               On the build's page, the warnings are listed under
+               `GNU C Compiler` and `CMake`. You can also check the
+               `Console Output` for the full warning logs.
+    * Actions: Does not detect compiler or CMake warnings.
+* There are static checker failures. All code checker failures must be fixed.
+    * Jenkins: Does not run static checkers.
+    * Actions: When there are code cherk failures, the build is marked red ❌.
+               On the build logs, the checker failures should be under the
+               `Code check` collapsible.
+* There are infrastructure failures. These should be reported to the build farmer.
```

Comment:
How do external contributors report to the build farmer? Should they create/add to an issue? If so, where can this issue be created/found?